src/pymor/models/iosys.py

```diff
@@ -554,7 +566,52 @@ def poles(self, mu=None):
         E = None if isinstance(E, IdentityOperator) else to_matrix(E, format='dense')
         return spla.eigvals(A, E)
 
+    def poles(self, mu=None):
+        """Compute system poles.
+
+        .. note::
+            Assumes the systems is small enough to use a dense eigenvalue solver.
+
+        Parameters
+        ----------
+        mu
+            |Parameter values| for which to compute the systems poles.
+
+        Returns
+        -------
+        One-dimensional |NumPy array| of system poles.
+        """
+        poles = self.presets['poles'] if 'poles' in self.presets else self._poles(mu=mu)
+        assert isinstance(poles, np.ndarray) and len(poles) == (self.A.source.dim)
+
+        return poles
+
     @cached
+    def _gramian(self, typ, mu=None):
+        A = self.A.assemble(mu)
+        B = self.B
+        C = self.C
+        E = self.E.assemble(mu) if not isinstance(self.E, IdentityOperator) else None
+        options_lrcf = self.solver_options.get('lyap_lrcf') if self.solver_options else None
+        options_dense = self.solver_options.get('lyap_dense') if self.solver_options else None
+        solve_lyap_lrcf = solve_cont_lyap_lrcf if self.sampling_time == 0 else solve_disc_lyap_lrcf
+        solve_lyap_dense = solve_cont_lyap_dense if self.sampling_time == 0 else solve_disc_lyap_dense
+
+        if typ == 'c_lrcf' and 'c_dense' in self.presets:
+            return self.A.source.from_numpy(_chol(self.presets['c_dense']).T)
+        elif typ == 'o_lrcf' and 'o_dense' in self.presets:
+            return self.A.source.from_numpy(_chol(self.presets['o_dense']).T)
```

Comment:
The other way around is also technically possible (`typ == 'c_dense' and 'c_lrcf' in self.presets` and `typ == 'o_dense' and 'o_lrcf' in self.presets`).